### ISSUE TITLE
Attach binaries to Github releases

### ISSUE THREAD

**Issue by g4s8** [articulate/terraform-provider-okta#128]
It would be great if you attach binaries to Github releases, and add a shell script to install it in a `README.md` file, so it will be much easier to use on CI instances. You can use this tool if you want: https://goreleaser.com/ it can upload binaries to Github releases.

**Comment by quantumew** [comment]
Will do. I can hook up goreleaser in our CI system and just have it tag and release whenever new changes hit master. The release process for this is definitely subpar. Should get that up fairly soon

**Comment by salekseev** [comment]
I could help with a PR to enable releases but it would require a `GITHUB_TOKEN` environment variable with a valid GitHub OAuth token that has the `public_repo` or `repo` scope to upload assets.

If there was a pre-encrypted value in `.travis.yml` like described in https://docs.travis-ci.com/user/deployment/releases/#authenticating-with-an-oauth-token...

**Comment by quantumew** [comment]
I have time today, I will get this in before the EOD. Thanks for the reminder. Mostly want to do this since we have an internal release process I need to adhere to. Need to push the releases to S3 as well as Github.

**Comment by quantumew** [comment]
Ok released the latest binaries. Still have to finish automating and documenting this but should have time tonight to get that wrapped up.

**Comment by salekseev** [comment]
@quantumew There's a bit of a problem with releases that are attached. It seems each tarball contains 3 files:
```
terraform-provider-okta_2.18.0_darwin_amd64.tar.gz: OK
x LICENSE
x README.md
x terraform-provider-okta
```

It's common practice (by hashicorp and other community) for providers to be compressed using deflate (ZIP) method and contain a binary that is versioned as well as this is how terraform distinguishes between provider versions. For example `terraform-provider-okta_v2.18.0_x4` where x4 is a version of GRPC communication protocol (https://github.com/hashicorp/terraform/issues/15561).

**Comment by quantumew** [comment]
Silly me, I was in a rush and didn't check my work. Fixing that now. Yea you are right let me get it automated there for now and skip our internal build system. That is really what is blocking me from getting this in master.

**Comment by quantumew** [comment]
@salekseev Thanks for your cogent and helpful response. Thanks for your patience as well on this. This is corrected, with v2.20.0. Please make any corrections if you see anything off here https://github.com/articulate/terraform-provider-okta/blob/master/.goreleaser.yml. TIL about the protocol version, GTK. Also as an aside I need to make some corrections to our build, it appears Okta's API is occasionally returning an error with no content back. Will debug and correct that after hours.

**Comment by g4s8** [comment]
@quantumew I downloaded binary from release, and it's working, thanks!
